src/agni/server/manager/LoginManager.java

```diff
@@ -17,8 +17,10 @@ public LoginManager(InfoSender infoSender,
 
     @Override
     public void loginRequest(String ip, String user, String password) {
-        // TODO Auto-generated method stub
-        
+    	if (password == userDataGuard.getPassword(user)) 
```

Comment:
To authenticate, you would hash the password with the salt. Here is code to generate a hash and get a password hash.

```java
public String findHash(username, passwordAttempt)
    string salt = userdataguard.salt(username);
    keyspec spec = new pbekeyspec(passwordattempt.tochararray(), salt, 65536, 128);
    secretkeyfactory f = secretkeyfactory.getinstance("pbkdf2withhmacsha1");
    byte[] hash = f.generatesecret(spec).getencoded();
    base64.encoder enc = base64.getencoder();
    return enc.encodetostring(hash);
}
```